### PR TITLE
[infra-proxy-service] GET policyfile revisions API 

### PR DESCRIPTION
**Pull request by vinay033**
### :nut_and_bolt: Description: What code changed, and why?
Need to add the backend API to fetch the list of the policy file.

### :chains: Related Resources
https://github.com/chef/automate/issues/5309
### :+1: Definition of Done
  - I have added API to fetch the list of the policyfile.
### :athletic_shoe: How to Build and Test the Change
Steps to build:
```
rebuild components/infra-proxy-service
rebuild components/automate-gateway
```
if you don't have sample data then you need to add first
1.  Ensure Chef Infra Server is running. If it is not start it with `start_chef_server`
2. First you need to run the `infra_service_load_chef_repo`
3. Then run `converge_policyfile_chef_client`

Steps to test:

Add the chef-server test environment credentials and extract the SERVER and ORG IDs.
Policy group details API - 
request 
```
curl -sSkH "api-token: $(get_admin_token)"  https://a2-dev.test/api/v0/infra/servers/SERVER/orgs/ORG/policyfiles/NAME/revisions?pretty
```
response 
```
{
  "revisions": [
    {
      "revision_id": "24a6c87fbe5359b5191cc866ce969020f8050332ac94b6a2349ca71682f619f5"
    },
    {
      "revision_id": "9d48026004f474c97bf703589f5a6bfd660448e5c54a8a7eecc897d627e9fba8"
    }
  ]
}
```
### :white_check_mark: Checklist

**All PRs** must tick these:

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/infra-proxy-service/server/policyfiles.go [hunk 1]
```diff
@@ -187,3 +185,37 @@ func (s *Server) DeletePolicyfile(ctx context.Context, req *request.DeletePolicy
...
+		return nil, ParseAPIError(err)
+	}
+
+	return &response.PolicyfileRevisions{
+		Revisions: fromAPIIncludedPolicyfileRevisions(policyfileRevision),
+	}, nil
+}
+
+// fromAPIIncludedPolicyfileRevisions a response included policyfile revision
+func fromAPIIncludedPolicyfileRevisions(p chef.PolicyGetResponse) []*response.PolicyfileRevision {
```
Comment (kalroy) [inline components/infra-proxy-service/server/policyfiles.go:207]: We should add some unit tests for this considering that the response from server is going to change.

**Review by kalroy — CHANGES_REQUESTED** [review]
Add the unit test mentioned.

**Review by rickmarry — COMMENTED** [review]
in the instructions one of the steps is:
`3. Then run load_policy_files_sample_data`

this is what comes back when I run that command in hab:
```
[10][default:/src:0]# load_policy_files_sample_data
bash: load_policy_files_sample_data: command not found
```

**Comment by vinay033** [comment]
> in the instructions one of the steps is:
> `3. Then run load_policy_files_sample_data`
> 
> this is what comes back when I run that command in hab:
> 
> ```
> [10][default:/src:0]# load_policy_files_sample_data
> bash: load_policy_files_sample_data: command not found
> ```

updated its `converge_policyfile_chef_client`

**Comment by rickmarry** [comment]
> > in the instructions one of the steps is:
> > `3. Then run load_policy_files_sample_data`
> > this is what comes back when I run that command in hab:
> > ```
> > [10][default:/src:0]# load_policy_files_sample_data
> > bash: load_policy_files_sample_data: command not found
> > ```
> 
> updated its `converge_policyfile_chef_client`

thanks

**Review by rickmarry — APPROVED** [review]
this looks good thanks @vinay033!

**Review by kalroy — APPROVED** [review]

**Pull request merged**